File: client/src/main/java/xyz/beriholic/beeyes/client/entity/MachineInfo.java
package xyz.beriholic.beeyes.client.entity;

import lombok.Data;
import lombok.experimental.Accessors;

@Data
@Accessors(chain = true)
public class MachineInfo {
    String osArch;
    String osName;
    String osVersion;
    int osBitSize;
    String cpuName;
    int cpuCoreCount;
    double memorySize;
    double diskSize;
    double diskTotalSize;
    NetworkInterfaceInfo networkInterfaceInfo;
}

File: client-go/internel/model/machine_info.go
package model

type MachineInfo struct {
	OsArch               string               `json:"osArch"`
	OsName               string               `json:"osName"`
	OsVersion            string               `json:"osVersion"`
	OsBitSize            int                  `json:"osBitSize"`
	CpuName              string               `json:"cpuName"`
	CpuCoreCount         int                  `json:"cpuCoreCount"`
	MemorySize           float64              `json:"memorySize"`
	DiskSize             float64              `json:"diskSize"`
	DiskTotalSize        float64              `json:"diskTotalSize"`
	NetworkInterfaceInfo NetworkInterfaceInfo `json:"networkInterfaceInfo"`
}
